### PR TITLE
Explicitly fail fixtures with test_ prefix (fix #510)

### LINKED ISSUE #510: fixture with "test_" prefix should be raise an error
Hi,
I've got it from the `yield_fixture`, where there is exception:
``` ~/.venv/bin/slash:11
>        load_entry_point('slash==1.1.6', 'console_scripts', 'slash')()
 ~/.venv/local/lib/python2.7/site-packages/slash/core/fixtures/utils.py:108
>        @context.fixture.add_cleanup
E    AttributeError: 'NoneType' object has no attribute 'add_cleanup'
```
But also on regular fixture, the test loader consider it as additional test, which is not.

```
$ cat a.py 
import slash

@slash.fixture
def test_my_fixture():
    return

@slash.yield_fixture
def test_my_yield_fixture():
    yield

def test_my_test(test_my_fixture, test_my_yield_fixture):
    pass

$ slash run a.py 
3 tests collected   
a.py ..E  FAIL
===================================== Session Summary ======================================
== #3: a.py:test_my_yield_fixture ==========================================================
 - 1/1 E (2017-01-25 10:39:42 +02:00):   - - - - - - - - - - - - - - - - - - - - - - - - - -
 ~/.venv/bin/slash:11
>        load_entry_point('slash==1.1.6', 'console_scripts', 'slash')()
 ~/.venv/local/lib/python2.7/site-packages/slash/core/fixtures/utils.py:108
>        @context.fixture.add_cleanup
E    AttributeError: 'NoneType' object has no attribute 'add_cleanup'
== Session ended. 2 successful, 0 skipped, 0 failures, 1 errors. Total duration: 00:00:00 ==
```

### PR DESCRIPTION
**Pull request by ayalash**

### CODE REVIEW

**Comment by ayalash** [comment]
Should fix https://github.com/getslash/slash/issues/510

File: doc/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,7 @@ Changelog
 =========
 
 * :feature:`507` Test id can now be obtained via ``slash.context.test.id``
+* :bug:`510` Explicitly fail fixtures which name is valid for tests (currently: test_ prefix)
```
Comment (vmalloc) [inline doc/changelog.rst]: This should be a `510 major` in order to appear in the release notes of 1.2.0

**Comment by ayalash** [comment]
@vmalloc Fixed (changelog comment) & re-pushed

**Comment by vmalloc** [comment]
Thanks!

**Pull request merged**